File: 485. Max Consecutive Ones.cpp
class Solution
{
public:
    int findMaxConsecutiveOnes(vector<int> &nums)
    {
        int curr_cons = 0;
        int max_ = 0;
        for (int n : nums)
        {
            if (n == 0)
            {
                curr_cons = 0;
            }
            else
            {
                curr_cons += 1;
                if (curr_cons > max_)
                    max_ = curr_cons;
            }
        }
        return max_;
    }
};

File: 485. Max Consecutive Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int currMax = 0;
        int consecutive = 0;
        for (int num : nums) {
            if (num == 1) {
                consecutive++;
                currMax = Math.max(consecutive, currMax);
            } else {
                consecutive = 0;
            }
        }
        return currMax;
    }
}